src/esi/ExpatParser.cc

```diff
@@ -12,8 +12,28 @@
 
 #if USE_SQUID_ESI && HAVE_LIBEXPAT
 
+#include "base/RunnersRegistry.h"
 #include "esi/ExpatParser.h"
 
+static ESIParser::Register *prExpat = nullptr;
+
+class EsiExpatRr : public RegisteredRunner
+{
+public:
+    void finalizeConfig()
+    {
+        prExpat = new ESIParser::Register("expat", &ESIExpatParser::NewParser);
+    }
+
+    void finishShutdown()
+    {
+      delete prExpat;
+      prExpat = nullptr;
+    }
```

Comment:
Same problem as in https://github.com/squid-cache/squid/pull/954#discussion_r772620417. Please remove this override (and adjust the remaining code accordingly).

Assuming ESIParser::Register destructor itself is safe/working, the (adjusted) cleanup code should go into the EsiExpatRr destructor AFAICT.

Reply:
Done in d28e46a. And updated description to highlight that this is a significant change to shutdown behaviour.